### ISSUE TITLE
Clearer error message when an ES6 class extends an interface

### ISSUE THREAD

**Issue by MatrixFrog** [google/closure-compiler#1643]
https://jscomp-debug.googleplex.com/#input0%3D%252F**%2520%2540interface%2520*%252F%250Aclass%2520MyInterface%2520%257B%257D%250A%250Aclass%2520MyClass%2520extends%2520MyInterface%2520%257B%257D%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_SYMBOLS%3D1%26CHECK_TYPES%3D1%26LANG_IN_IS_ES6%3D1%26MISSING_PROPERTIES%3D1%26PRESERVE_TYPE_ANNOTATIONS%3D1%26PRETTY_PRINT%3D1%26TRANSPILE%3D1

Currently this gives two different errors, and one of them mentions `$jscomp.inherits` which is totally unhelpful and meaningless to the user. We should:
1. Only report one error
2. Make sure it's clear
3. (bonus) Suggest that the user probably meant to include an `@implements` annotation in their JSDoc, rather than writing `extends`

**Comment by MatrixFrog** [comment]
@Dominator008 want to take a stab at this?

**Comment by Dominator008** [comment]
```
input0:4: WARNING -  cannot extend this type; constructors can only extend constructors
class MyClass extends MyInterface {}
^

input0:4: WARNING - actual parameter 2 of $jscomp.inherits does not match formal parameter
found   : function (this:MyInterface): ?
required: Function
class MyClass extends MyInterface {}
^

0 error(s), 2 warning(s), 89.8% typed
```
I think both of them come from `TypeCheck`, which knows nothing about ES6. If we want to generate a meaningful and helpful error message from the transpilation pass would it be a third error (or first, in output order)?

**Comment by MatrixFrog** [comment]
Or the transpilation could add a little bit of metadata saying "this used
to be an ES6 class" which the typechecker could use to modify its error
messages. That could get unwieldy though. I'm not sure.

The first message is clearer IMO so just getting rid of the second somehow
would help quite a bit.

**Comment by blickly** [comment]
I don't think we want to complicate typechecking just to give a better message here.

@MatrixFrog Is the unqualified name the main case that we care about? If so, it might be easy to write a syntactic check based just on the @interface annotation in the JSDoc.

**Comment by MatrixFrog** [comment]
The problem is the interface might be in another file and we want
transpilation to work in a file-by-file way.

**Comment by Dominator008** [comment]
@MatrixFrog Smaller repo:
```js
/** @interface */ function Foo() {};
var /** Function */ x = Foo;
```
https://closure-compiler-debugger.appspot.com/#input0%3D%252F**%2520%2540interface%2520*%252F%2520function%2520Foo()%2520%257B%257D%253B%250A%250Avar%2520%252F**%2520Function%2520*%252F%2520x%2520%253D%2520Foo%253B%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_TYPES%3D1
